server/komodo-rest/src/main/java/org/komodo/rest/relational/KomodoRestUriBuilder.java

```diff
@@ -649,6 +705,59 @@ public URI vdbPermissionChildUri(LinkType linkType, final Properties settings) {
     }
 
     /**
+     * @param parentUri the parent URI
+     * @param dataserviceName the dataservice name
+     * @return the URI for the given dataservice
+     */
+    public URI dataserviceUri(final URI parentUri, final String dataserviceName) {
+        return UriBuilder.fromUri(parentUri).path(dataserviceName).build();
+    }
+
+    /**
+     * @param linkType
+     *        the type of URI being created (cannot be <code>null</code>)
+     * @param settings
+     *        configuration settings for this uri
+     * @return the dataservice URI for the specified dataservice (never <code>null</code>)
+     */
+    public URI dataserviceUri(final LinkType linkType, final Properties settings) {
+        ArgCheck.isNotNull(linkType, "linkType"); //$NON-NLS-1$
+        ArgCheck.isNotNull(settings, "settings"); //$NON-NLS-1$
+
+        URI parentUri = dataserviceParentUri(settings);
+        URI result = null;
+
+        switch (linkType) {
+            case SELF:
+            {
+                String dataserviceName = dataserviceName(settings);
+                result = dataserviceUri(parentUri, dataserviceName);
+                break;
+            }
+            case PARENT:
+            {
+                result = parentUri;
+                break;
+            }
+            case IMPORTS:
+            case MODELS:
+            case TRANSLATORS:
+            case DATA_ROLES:
+            case VDBS:
```

Comment:
Did you really want these to cascade into VDBS? Probably should cascade into default?